src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,423 @@ title: Builder.io & Astro
+---
+export async function getStaticPaths() {
+  const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+  const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+
+  const { results: posts } = await fetch(
+    `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams(
+      {
+        apiKey: builderAPIpublicKey,
+        fields: ["data.slug", "data.title"].join(","),
+        cachebust: "true",
+      }
+    ).toString()}`
+  )
+    .then((res) => res.json())
+    .catch
+    // ...catch some errors...);
+    ();
+
+  return [
+    ...posts.map(({ data: { slug, title } }) => [
+      {
+        params: { slug },
+        props: { title },
+      },
+    ]),
+  ];
+}
+---
+```
+
+
+Next, add the following content:
+
+```astro title="src/pages/posts/[slug].astro" ins={30-49}
+---
+export async function getStaticPaths() {
+  const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+  const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+
+  const { results: posts } = await fetch(
+    `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams(
+      {
+        apiKey: builderAPIpublicKey,
+        fields: ["data.slug", "data.title"].join(","),
+        cachebust: "true",
+      }
+    ).toString()}`
+  )
+    .then((res) => res.json())
+    .catch
+    // ...catch some errors...);
+    ();
+
+  return [
+    ...posts.map(({ data: { slug, title } }) => [
+      {
+        params: { slug },
+        props: { title },
+      },
+    ]),
+  ];
+}
+
+const { slug } = Astro.params;
+const { title } = Astro.props;
+
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+// Builder's API requires this field but for this use case the URL doesn't seem to matter - the API returns the same HTML
+const encodedUrl = encodeURIComponent("moot");
+
+const { html: postHTML } = await fetch(
+  `https://cdn.builder.io/api/v1/qwik/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    url: encodedUrl,
+    "query.data.slug": slug,
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+```
+
+The `slug` returned from `getStaticPaths` is used to fetch the corresponding post. The HTML that Builder sends back is made of the blocks you put on the page in Builder's editor.
+
+:::caution
+This code relies on the fact that each content entry in the `blogpost` model has non-empty fields called `slug` and `title`. If you didn't create a fresh model for this guide, make sure that all content entries have these fields filled, including those that you started with.
+:::
+
+:::note
+The variables `builderModel` and `builderAPIpublicKey` need to be created twice, since [`getStaticPaths` runs in its own isolated scope](/en/reference/api-reference/#getstaticpaths).
+:::
+
+
+Finally, render the HTML:
+
+```astro title="src/pages/posts/[slug].astro" ins={52-99}
```

Comment:
``````suggestion
This file must contain:
- A `getStaticPaths()`](/en/reference/api-reference/#getstaticpaths) function to fetch `slug` information from Builder and create a static route for each blog post.
- A `fetch()` to the Builder API using the `slug` identifier to return post content and metadata (e.g. a `title`).
- A `<Fragment />` in the template to render the post content as HTML.

:::note
The variables `builderModel` and `builderAPIpublicKey` need to be created twice, since `getStaticPaths()` runs in its own isolated scope.
:::

```astro title="src/pages/posts/[slug].astro"
``````